### DISCUSSION TITLE
DB not working after superset upgrade.

### DISCUSSION THREAD

**Discussion by anujpareeksynoriq** [apache/superset#17994] [Q&A / Help]
Database is not working after upgrading to the latest version.

#### How to reproduce the bug
Upgraded the superset using this way:

1. docker-compose down
2. cd superset/
3. git pull origin master
4.  sudo docker-compose up

Is there any step, any service killing any thing I need to do ? My other database that I was previously using with superset is running on 5432. I had created important charts on it. I do not want to lose them.

Here are the logs that I got after running  **_docker-compose up_**

**LOGS:** 

WARNING: The CYPRESS_CONFIG variable is not set. Defaulting to a blank string.
Creating network "superset_default" with the default driver
Creating superset_cache ...
Creating superset_db    ...
Creating superset_db    ... error

Creating superset_cache ... done
d proxy: listen tcp4 127.0.0.1:5432: bind: address already in use
Creating superset_websocket ... done

ERROR: for db  Cannot start service db: driver failed programming external connectivity on endpoint superset_db (8b5e8a3b45b25c603ec056544757f292f28b26669abc103471512d2310af759f): Error starting userland proxy: listen tcp4 127.0.0.1:5432: bind: address already in use
ERROR: Encountered errors while bringing up the project.

### Expected results

what you expected to happen.

The docker compose should start normally.

what actually happens.

The docker compose shows some error (shared below.) 

After running **_docker-compose -f docker-compose-non-dev.yml up_** , The application is starting normally but I am unable to see the databases. 

If the database migration is needed, then please suggest how to do it.
**REFERRING TO THIS:** https://superset.apache.org/docs/installation/upgrading-superset#docker-compose
#### Screenshots

![image](https://user-images.githubusercontent.com/82150530/142990277-143bcb8d-72fb-4a7d-a9e5-0fb5d8fd2481.png)

### Environment

(please complete the following information):

- browser type and version:
- superset version: 3.7
- python version: 3.8
- node.js version: `node -v`
- any feature flags active:

### Checklist

Make sure to follow these steps before submitting your issue - thank you!

- [x] I have checked the superset logs for python stacktraces and included it here as text if there are any.
- [x] I have reproduced the issue with at least the latest released version of superset.
- [x] I have checked the issue tracker for the same issue and I haven't found one similar.

**Comment by anujpareeksynoriq** [comment]
**[SOLUTION]:** The issue was because the default postgres that comes with superset was trying to run on 5432. But I had a postgres server already running on that port. I restarted my database on a different port and things worked as expected.

**[IMPORTANT QUESTION]:** Initially when I setup super set,  I had my custom postgres running on 5432 only, So how was it working before taking the latest superset pull?  
2. How do I change the superset default postgres service port number ? Where is this config written?

**Reply by eschutho** [reply]
@anujpareeksynoriq I've run into conflicts on earlier versions of Superset with a different database running on port 5432, so I don't know why it was working for you before but not now, i.e., I'm not aware of any recent changes. But to change the setting you would update your docker config here: https://github.com/apache/superset/blob/master/docker-compose.yml#L47
